### DISCUSSION TITLE
disk management when replica goes offline

### DISCUSSION THREAD

**Discussion by dberardo-com** [patroni/patroni#3143] [Q&A]
is it possible to prevent the wal directory to explode whenever a replica goes offline for a longer period of time? like imposing a hard limit on the wal dir size

**Comment by CyberDem0n** [comment]
pg_wal may explode due to different reasons, and stale replication slots is only one of them.
When replica is offline longer than `member_slots_ttl` (default 30min), Patroni v4.0+ will remove its replication slot it automatically created earlier.
Before v4.0 slot is removed after `ttl` (default 30s).

From the Postgres side there are a few GUCs (max_wal_size, max_slot_wal_keep_size, wal_keep_size), but they implement only soft limits.

**Comment by dberardo-com** [comment]
i have the following situation:

![image](https://github.com/user-attachments/assets/bf6523b0-e9c4-47a0-8b45-1077d0aead3c)
and 

![image](https://github.com/user-attachments/assets/b8723919-c589-4b4a-b49f-4a34028873ef)

but the pg_wal directory keeps increasing.

i believe this is not "patroni's" fault, but rather pgbackrest's one, which i am using for bakcups. Could it be that the pg_wal folder keeps increasing even if there is no replcation slot active because of pgbakcrest pushing the archive ?

---

NOTE: i havent set max_slot_wal_keep_size

**Reply by CyberDem0n** [reply]
You have to figure out why replicas are lagging. That is, start from checking postgres logs.
